projects/core/src/lib/components/feature-info/state/feature-info.selectors.ts

```diff
@@ -135,3 +136,41 @@ export const selectFeatureInfoLayerListItems = createSelector(
     }));
   },
 );
+
+export const selectCurrentFeatureForEdit = createSelector(
+  selectFeatureInfoFeatures,
+  selectSelectedFeatureInfoLayer,
+  selectFeatureInfoMetadata,
+  (features, selectedLayer, featureInfoMetadata) => {
+    if (!selectedLayer || !selectedLayer.selectedFeatureId) {
+      return null;
+    }
+    const feature = features.find(f => f.__fid === selectedLayer.selectedFeatureId);
+    if (feature && featureInfoMetadata) {
+      const geometryAttributeName = featureInfoMetadata.find(m => m.layerId === feature.layerId && m.type === AttributeType.GEOMETRY)?.key;
+      if (!geometryAttributeName) {
+        return null;
+      }
+      const newAttributes = { ...feature.attributes };
+      newAttributes[geometryAttributeName] = feature.geometry;
+      const newFeature: FeatureInfoFeatureModel = {
+        ...feature,
+        attributes: newAttributes,
+      };
+      const filteredMetadata = featureInfoMetadata.filter(m => m.layerId === feature.layerId);
+      const newColumnMetadata = [
+        ...filteredMetadata,
+        {
+          layerId: feature.layerId,
+          key: geometryAttributeName,
+          type: AttributeType.GEOMETRY,
+        },
+      ];
```

Comment:
When fixing one comment please look at the entire method.  This can be simplified by first defining `filteredMetadata` (filtering featureInfoMetadata only once), newColumnMetadata is also not necessary anymore, now it even has a duplicate geometry column.